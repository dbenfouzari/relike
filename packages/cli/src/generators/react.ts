import { paramCase, snakeCase } from "change-case";
import fs from "fs";
import path from "path";

import { replace, replaceFileName } from "../utils/template";

/**
 * Generates component index file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateIndexFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateIndexFile = (key: string, componentName: string) => {
  const templateIndexPath = path.resolve(__dirname, "../templates/index/index.ts.txt");

  const templateIndex = fs.readFileSync(templateIndexPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateIndexPath, key, componentName)),
    content: replace(templateIndex, key, componentName, paramCase),
  };
};

/**
 * Generates component style file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateStyleFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateStyleFile = (key: string, componentName: string) => {
  const templateStylePath = path.resolve(__dirname, "../templates/react-style/__name__.module.scss.txt");

  const templateStyle = fs.readFileSync(templateStylePath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateStylePath, key, componentName)),
    content: replace(templateStyle, key, componentName, snakeCase),
  };
};

/**
 * Generates component story file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateStoryFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateStoryFile = (key: string, componentName: string) => {
  const templateStoryPath = path.resolve(__dirname, "../templates/react-story/__name__.stories.tsx.txt");

  const templateStory = fs.readFileSync(templateStoryPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateStoryPath, key, componentName)),
    content: replace(templateStory, key, componentName),
  };
};

/**
 * Generates component test file
 *
 * @param   {string} key           The key to replace
 * @param   {string} componentName The component name
 * @example
 * generateTestFile("name", name);
 * @returns {any}                  File path and content.
 */
export const generateTestFile = (key: string, componentName: string) => {
  const templateTestPath = path.resolve(__dirname, "../templates/react-test/__name__.test.tsx.txt");

  const templateTest = fs.readFileSync(templateTestPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateTestPath, key, componentName)),
    content: replace(templateTest, key, componentName),
  };
};

/**
 * Generates component file
 *
 * @param   {string}  key           The key to replace
 * @param   {string}  componentName The component name
 * @param   {boolean} hasStyle      Does the component have style
 * @example
 * generateComponentFile("name", name, flags.style);
 * @returns {any}                   File path and content.
 */
export const generateComponentFile = (key: string, componentName: string, hasStyle: boolean) => {
  const templateComponentPath = hasStyle
    ? path.resolve(__dirname, "../templates/react-component/__name__(if:style).tsx.txt")
    : path.resolve(__dirname, "../templates/react-component/__name__.tsx.txt");

  const templateComponent = fs.readFileSync(templateComponentPath, { encoding: "utf-8" });

  return {
    path: path.basename(replaceFileName(templateComponentPath, key, componentName)),
    content: replace(templateComponent, key, componentName),
  };
};
