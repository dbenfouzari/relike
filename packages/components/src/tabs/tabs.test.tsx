import { fireEvent, render, screen } from "@testing-library/react";

import Tabs from "./tabs";

const children = [
  <Tabs.Item key="1" name="Tab 1">
    Content of Tab 1
  </Tabs.Item>,
  <Tabs.Item key="2" name="Tab 2">
    Content of Tab 2
  </Tabs.Item>,
  <Tabs.Item key="3" name="Tab il Ã©tait un petit navire">
    Content of Tab 3
  </Tabs.Item>,
  <Tabs.Item key="4" name="Tab 4">
    Content of Tab 4
  </Tabs.Item>,
];

describe("Tabs", () => {
  it("should render successfully", () => {
    render(<Tabs>{children}</Tabs>);

    expect(screen.getByRole("tablist")).toMatchSnapshot();
  });

  it("should work with defaultActiveKey", () => {
    render(<Tabs defaultActiveKey="3">{children}</Tabs>);

    expect(screen.getByRole("tablist")).toMatchSnapshot();
  });

  it("should handle click", () => {
    render(<Tabs defaultActiveKey="3">{children}</Tabs>);

    fireEvent.click(screen.getByTestId(2));
  });
});
