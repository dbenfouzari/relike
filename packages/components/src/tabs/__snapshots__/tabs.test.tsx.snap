// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tabs should render successfully 1`] = `
<div
  class="wrapper wrapper-0-2-1 wrapper-d0-0-2-2"
  role="tablist"
>
  <button
    class="item item-0-2-3 item-d0-0-2-4 item__active"
    data-testid="0"
    role="tab"
  >
    Tab 1
  </button>
  <button
    class="item item-0-2-3 item-d1-0-2-5"
    data-testid="1"
    role="tab"
  >
    Tab 2
  </button>
  <button
    class="item item-0-2-3 item-d2-0-2-6"
    data-testid="2"
    role="tab"
  >
    Tab il était un petit navire
  </button>
  <button
    class="item item-0-2-3 item-d3-0-2-7"
    data-testid="3"
    role="tab"
  >
    Tab 4
  </button>
  <div
    class="bar"
    style="width: 0px; left: 0px;"
  />
</div>
`;

exports[`Tabs should work with defaultActiveKey 1`] = `
<div
  class="wrapper wrapper-0-2-1 wrapper-d1-0-2-8"
  role="tablist"
>
  <button
    class="item item-0-2-3 item-d4-0-2-9"
    data-testid="0"
    role="tab"
  >
    Tab 1
  </button>
  <button
    class="item item-0-2-3 item-d5-0-2-10"
    data-testid="1"
    role="tab"
  >
    Tab 2
  </button>
  <button
    class="item item-0-2-3 item-d6-0-2-11 item__active"
    data-testid="2"
    role="tab"
  >
    Tab il était un petit navire
  </button>
  <button
    class="item item-0-2-3 item-d7-0-2-12"
    data-testid="3"
    role="tab"
  >
    Tab 4
  </button>
  <div
    class="bar"
    style="width: 0px; left: 64px;"
  />
</div>
`;
